modules/ROOT/pages/dataweave-types.adoc

```diff
@@ -169,23 +177,130 @@ Where the [n] is replaced by the value for each of the date and time elements th
 - `M` is the minute designator (e.g. `|PT1M|`)
 - `S` is the second designator (e.g. `|PT1S|`)
 
-.Transform
+`DatePeriod` is useful for date manipulation, such as addition or subtraction. +
+The following example shows how to subtract one year from a date using `DatePeriod`. The return type is a new `Date`:
+
+.Source
+[source,dataweave,linenums]
+----
+output application/dw
+---
+|2003-10-01| - |P1Y|
+----
+
+.Output
+[source,dataweave,linenums]
+----
+|2002-10-01|
+----
+
+`Duration` is useful as a result of a `Date` to `Date` subtraction so the amount of time between those two dates can be inferred. +
+The following example shows how the result of subtracting two `Date` returns a `Duration`:
+
+.Source
+[source,dataweave,linenums]
+----
+output application/dw
+---
+|2003-11-01| - |2003-10-01|
+----
+
+.Output
+[source,dataweave,linenums]
+----
+|PT744H|
+----
+
+==== Period Coercion
+
+As `DatePeriod` is date-based it can be coerced to Number with different date based units such as `years` or `months`. +
+The following example shows how to coerce a `DatePeriod` to a `Number` by using different units:
+
+.Source
+[source,dataweave,linenums]
+----
+output application/json
+---
+{
+ years:  |P1Y12M| as Number {unit: "years"},
+ months: |P8Y12M| as Number {unit: "months"}
+}
+----
+
+.Output
+[source,json,linenums]
+----
+{
+  "years": 2,
+  "months": 108
+}
+----
+
+==== Duration Coercion
+
+Because `Duration` is time-based, this type can be coerced to a `Number` with different time-based units, such as `nanos`, `milliseconds`, `seconds`, `hours`, and `days`.
```

Comment:
```suggestion
Because a `Duration` value is time-based, it can be coerced to a `Number` value with different time-based units, such as `nanos`, `milliseconds`, `seconds`, `hours`, and `days`.
```

Reply:
If we are including these coersion sections, do we really need to mention "conversion" in the period definitions?